sympy/matrices/matrices.py

```diff
@@ -1677,6 +1678,40 @@ def limit(self, *args):
 # https://github.com/sympy/sympy/pull/12854
 class MatrixDeprecated(MatrixCommon):
     """A class to house deprecated matrix methods."""
+    def _array_dot(self, b):
```

Comment:
I think we should rename this one more time to `_legacy_array_dot`, so it's extra clear to people looking at the code.